src/core/config.c

```diff
@@ -423,6 +484,27 @@ static struct config_list *parse_file(FILE *file)
         return NULL;
 }
 
+void destroy_config_list(struct config_list *list)
+{
+        for (size_t i = 0; i < list->length; ++i) {
+                destroy_config_value(list->values[i]);
+        }
+
+        free(list);
+}
+
+void destroy_config_value(struct config_value *value)
+{
+        if (value->type == NUMBER) {
```

Comment:
swap this conditional to look for `value->type == STRING` and just free the value to avoid free'ing `value` twice